src/docker-in-docker/install.sh

```diff
@@ -406,6 +420,21 @@ dockerd_start="AZURE_DNS_AUTO_DETECTION=${AZURE_DNS_AUTO_DETECTION} DOCKER_DEFAU
 
     # Start docker/moby engine
     ( dockerd $CUSTOMDNS $DEFAULT_ADDRESS_POOL > /tmp/dockerd.log 2>&1 ) &
+
+    # Wait for API to be listening on unix socket.
+    echo "Waiting for 'docker-in-docker' to be ready..."
```

Comment:
Question: Is this change specifically needed for docker to work with bookworm? or is it docker in general?

Reply:
And i suspect could also be the cause of some failures in the wild